cmd/k8s-deployment-scaler/main.go

```diff
@@ -0,0 +1,354 @@
+package main
+
+import (
+	"context"
+	"crypto/tls"
+	"crypto/x509"
+	"encoding/json"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+
+	appsv1 "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/client-go/informers"
+	"k8s.io/client-go/kubernetes"
+	"k8s.io/client-go/rest"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/tools/clientcmd"
+)
+
+// Define the global clientset variable using kubernetes.Interface
+var clientset kubernetes.Interface
+
+// DeploymentInfo stores the relevant information about a Deployment
+type DeploymentInfo struct {
+	Replicas   int32
+	Deployment *appsv1.Deployment
+}
+
+var deploymentInformer cache.SharedIndexInformer
+
+// main initializes and runs the server, setting up TLS configuration, handling graceful shutdown, and defining HTTP routes.
+func main() {
+	// Initialize Kubernetes client
+	var config *rest.Config
+	var err error
+
+	if os.Getenv("KUBERNETES_SERVICE_HOST") != "" {
+		config, err = rest.InClusterConfig()
+	} else {
+		config, err = clientcmd.BuildConfigFromFlags("", os.Getenv("KUBECONFIG"))
+	}
+
+	if err != nil {
+		log.Fatalf("Error building kubeconfig: %v", err)
+	}
+
+	clientset, err = kubernetes.NewForConfig(config)
+	if err != nil {
+		log.Fatalf("Error creating Kubernetes client: %v", err)
+	}
+
+	// Set up deployment informer to watch for changes in deployments
+	factory := informers.NewSharedInformerFactory(clientset, time.Minute*10)
+	deploymentInformer = factory.Apps().V1().Deployments().Informer()
```

Comment:
Usage of the lister might provide a better UX for you instead of the informer.

Reply:
The code now uses the lister.